### PR TITLE
FE: Profile Details page

### LINKED ISSUE #20: FE: Profile Details page
Each Profile should have its own dedicated page that outlines their profile in detail

### PR DESCRIPTION
**Pull request by rizperdana**
Fixes #20

### What this PR does (required):
- This MR will add profile detail skeleton

### Screenshots / Videos (front-end only):
Screenshot 2022-01-14 220339

### Any information needed to test this feature (required):
- `npm run start`
- login / register as user
- once you get in the dashboard, navigate your url to `/profile/detail`

### Any issues with the current functionality (optional):
- It's already responsive, but i need more input regarding this

### CODE REVIEW

**Review by sundayezeilo — COMMENTED** [review]
Good work with this one.

File: client/package.json [hunk 1]
```diff
@@ -5,6 +5,10 @@
   "dependencies": {
     "@emotion/react": "^11.6.0",
     "@emotion/styled": "^11.6.0",
+    "@material-ui/core": "^4.12.3",
+    "@material-ui/icons": "^4.11.2",
+    "@material-ui/lab": "^4.0.0-alpha.60",
+    "@material-ui/pickers": "^3.3.10",
```
Comment (moffatethan) [inline client/package.json]: You've installed v4 of Material UI, we're on v5 of Material UI. Please remove these packages.

File: client/package.json [hunk 2]
```diff
@@ -20,6 +24,7 @@
     "clsx": "^1.1.1",
     "date-fns": "^2.28.0",
     "eslint": "^7.26.0",
+    "faker": "^5.5.3",
```
Comment (moffatethan) [inline client/package.json]: I don't even see this being used anywhere so remove?

File: client/package.json [hunk 3]
```diff
@@ -55,6 +60,7 @@
     ]
   },
   "devDependencies": {
+    "@types/faker": "^5.5.3",
```
Comment (moffatethan) [inline client/package.json]: Same with this one too 😄

File: client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      time: number.toString() + ':00 PM',
+      value: number,
+    };
+  } else {
+    return {
+      time: (number - 12).toString() + ':00 PM',
+      value: number,
+    };
+  }
+});
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx]: Wonder if there's a way to make this a bit more intuitive? I'm going to think on this more, if you could think as well that would be excellent!

File: client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+const BookingForm = (): JSX.Element => {
+  const classes = useStyles();
+
+  const bookForm = useFormik({
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx:28]: Good job using Formik here, could we also add in form validation with Yup schemas? Formik has great support for Yup.

File: client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            ></OutlinedInput>
+            <Select
+              variant="outlined"
+              name="startTime"
+              className={classes.profileDetailsFormSelect}
+              value={bookForm.values.startTime}
+              onChange={bookForm.handleChange}
+            >
+              <MenuItem value="placeholder" disabled>
+                start time
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx]:
```suggestion
                Start time
```

File: client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+            ></OutlinedInput>
+            <Select
+              variant="outlined"
+              name="endTime"
+              className={classes.profileDetailsFormSelect}
+              value={bookForm.values.endTime}
+              onChange={bookForm.handleChange}
+            >
+              <MenuItem value="placeholder" disabled>
+                end time
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx]:
```suggestion
                End time
```

File: client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                <MenuItem value={time.value} key={time.value}>
+                  {time.time}
+                </MenuItem>
+              ))}
+            </Select>
+          </Grid>
+        </Box>
+
+        <Button type="submit" className={classes.requestFormButton}>
+          send request
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx]:
```suggestion
          Send Request
```

File: client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx [hunk 9]
```diff
@@ -0,0 +1,125 @@
+import useStyles from './useStyles';
+import { Box, Typography, Select, MenuItem, OutlinedInput, Grid, Button, Card } from '@material-ui/core';
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/BookingForm/BookingForm.tsx]: You aren't using the right material UI version here, so please update this to MUI v5.

File: client/src/pages/ProfileDetail/ProfileDetail.tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+            <CardMedia
+              className={classes.backgroundImg}
+              image={profile.backgroundImg}
+              title="Profile background image"
+            />
+            <CardContent className={classes.topCardContent}>
+              <CardMedia className={classes.userImg} image={profile.imageImg} title="Profile image" />
+              <Typography className={classes.userNameText}>{profile.name}</Typography>
+              <Typography className={classes.userTitleText}>{profile.aboutMe}</Typography>
+              <Box marginTop="20px" display="flex">
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/ProfileDetail.tsx]: For MUI v5, let's use `sx` for this.

File: client/src/pages/ProfileDetail/ProfileDetail.tsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  ],
+};
+
+const ProfileDetail = (): JSX.Element => {
+  const classes = useStyles();
+
+  return (
+    <Grid container className={classes.rootContainer} justify="space-evenly">
+      <Grid item xs={12} sm={12} md={7} lg={7}>
+        <Box margin="0 auto" display="grid">
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/ProfileDetail.tsx]: For MUI v5, let's use `sx` for these properties.

File: client/src/pages/ProfileDetail/ProfileDetail.tsx [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+              <CardMedia className={classes.userImg} image={profile.imageImg} title="Profile image" />
+              <Typography className={classes.userNameText}>{profile.name}</Typography>
+              <Typography className={classes.userTitleText}>{profile.aboutMe}</Typography>
+              <Box marginTop="20px" display="flex">
+                <RoomIcon className={classes.userLocationIcon} />
+                <Typography className={classes.userLocationText}>{profile.location}</Typography>
+              </Box>
+            </CardContent>
+            <CardContent className={classes.bottomCardContent}>
+              <Box marginTop="10px" width="100%" textAlign="left" marginBottom="30px">
```
Comment (moffatethan) [inline client/src/pages/ProfileDetail/ProfileDetail.tsx]: As mentioned before, MUI v5 has `sx` so let's use that for these properties.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Thanks for doing this, some comments on quality and quick changes. I noticed the colouring in the UI is not even the right colour compared to the overall app. Let's fix that. Also **please uninstall v4 of MUI and continue using v5.**

While you have a good initial responsiveness on the UI. It's not properly responsive, take a look at my screenshot on my laptop. It's very squished and looks like content is chopped off. On tablet this is happening too.

Mobile:
Screen Shot 2022-01-14 at 5 42 54 PM

Tablet:
Screen Shot 2022-01-14 at 5 57 38 PM

**Comment by rizperdana** [comment]
Hi @moffatethan i made some revision for this 
image

also revamping those responsive element
image

additionally i fix those weird date time picker function i made last time by using simplified version of native datepicker
image


TODO: i havent done fixing those form validation you suggesting, i will looking forward to it after i fixed my faulty hard drive here

**Pull request closed without merging**